.github/workflows/post-apiview.yml

```
@@ -0,0 +1,32 @@
+name: After APIView
+
+on:
+  check_run:
+    types: [completed]
+    
+permissions:
+  issues: write
+  contents: read
+
+jobs:
+  post-apiview:
+    name: After APIView
+    runs-on: ubuntu-24.04
+    if: |
+      ${{ github.event.check_run.check_suite.app.name == 'Azure Pipelines' && 
+      ( contains(github.event.check_run.name, 'APIView') ||
+      contains(github.event.check_run.name, '(SDK Generation') ||
+      contains(github.event.check_run.name, 'Azure.azure-rest-api-specs-pipeline') ) }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          sparse-checkout: 'eng/common'
+    
+      - name: Create APIView Comment on PR
+        run: |
+          . "${{ github.workspace }}/eng/common/scripts/Helpers/ApiView-Helpers.ps1"
+          Set-ApiViewCommentForRelatedIssues -HeadCommitish ${{ github.event.check_run.head_sha }} -AuthToken $env:AUTH_TOKEN
```

```suggestion
          Set-ApiViewCommentForRelatedIssues -HeadCommitish ${{ github.event.check_run.head_sha }}
```

If `$env:AUTH_TOKEN` is substituted for the value of the token on the CLI, I think this might have the same problem as the previous code.  I think you need to update `ApiView-Helpers.ps1` to load the env var directly from a fixed name `GITHUB_TOKEN`.  We don't want to print the `GITHUB_TOKEN` in any way, via logging, if a CLI gets printed, etc.

Or, if you really need to support env vars of arbitrary name, you could pass the **name** of the env var on the command-line, but not it's value.

@weshaggard: Do you agree?  You may know more about this than me.

---

@mikeharder doesn't Github Actions automatically mark the GITHUB_TOKEN as a secret and as such will cause it to be starred out at least in the GH logs? I would expect it should, and if so, what were we trying to fix by moving this to an env variable? Where is it getting logged?

As for a general pattern I think we have a mix of things being set as env or passed as a parameter but either way these values should be marked as secrets so they should be starred out in the normal logs.

---

For defense-in-depth, I think it's better to never even attempt to write the secret to the log, rather than rely on the log-redacting feature.  I thought one of the reasons to prefer env vars over CLI arguments, is the former are much less likely to be logged automatically.

---

I don't have a concern with using environment variables, although if we do use them we should just default the parameter to the env variable (i.e. in the script when defining variable `$AuthToken = $env:AUTH_TOKEN`) so that we have an easy way to override it if needed.

---

Also, if we are only using this for the gh cli we could use the wellknown variable `GH_TOKEN` for that.

---

We probably should update the code at https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/Invoke-GitHubAPI.ps1#L5 to use `GH_TOKEN` if the token parameter it is passed is null. That way we can use that environment variable consistently and not have this on the command line.